src/main/java/org/broadinstitute/consent/http/service/DatasetRegistrationService.java

```diff
@@ -0,0 +1,426 @@
+            throw new IllegalArgumentException("Invalid DAC id");
+        }
+
+        Map<String, BlobId> uploadedFileCache = new HashMap<>();
+
+        List<DatasetServiceDAO.DatasetInsert> datasetInserts = new ArrayList<>();
+        for (int consentGroupIdx = 0; consentGroupIdx < registration.getConsentGroups().size(); consentGroupIdx++) {
+            datasetInserts.add(createDatasetInsert(registration, user, files, uploadedFileCache, consentGroupIdx));
+        }
+
+        List<Integer> createdDatasetIds = datasetServiceDAO.insertDatasets(datasetInserts);
+        return datasetDAO.findDatasetsByIdList(createdDatasetIds);
+    }
+
+    private BlobId uploadFile(FormDataBodyPart file) throws IOException {
+        String mediaType = file.getContentDisposition().getType();
+
+        return gcsService.storeDocument(
+                file.getValueAs(InputStream.class),
+                mediaType,
+                UUID.randomUUID());
+    }
+
+    private DatasetServiceDAO.DatasetInsert createDatasetInsert(DatasetRegistrationSchemaV1 registration,
+                                                                User user,
+                                                                Map<String, FormDataBodyPart> files,
+                                                                Map<String, BlobId> uploadedFileCache,
+                                                                Integer consentGroupIdx) throws IOException {
+        ConsentGroup consentGroup = registration.getConsentGroups().get(consentGroupIdx);
+
+        List<DatasetProperty> props = convertRegistrationToDatasetProperties(registration, consentGroupIdx);
+        DataUse dataUse = generateDataUseFromConsentGroup(consentGroup);
+        List<FileStorageObject> fileStorageObjects = uploadFiles(files, uploadedFileCache, consentGroupIdx, user);
+
+        return new DatasetServiceDAO.DatasetInsert(
+                consentGroup.getConsentGroupName(),
+                registration.getDataAccessCommitteeId(),
+                dataUse,
+                user.getUserId(),
+                props,
+                fileStorageObjects
+        );
+    }
+
+    private DataUse generateDataUseFromConsentGroup(ConsentGroup group) {
+        DataUse dataUse = new DataUse();
+
+        dataUse.setCollaboratorRequired(group.getCol());
+        dataUse.setDiseaseRestrictions(group.getDiseaseSpecificUse());
+        dataUse.setEthicsApprovalRequired(group.getIrb());
+        dataUse.setGeneralUse(group.getGeneralResearchUse());
+        dataUse.setGeographicalRestrictions(group.getGs());
+        dataUse.setGeneticStudiesOnly(group.getGso());
+        dataUse.setHmbResearch(group.getHmb());
+        dataUse.setPublicationMoratorium(Objects.nonNull(group.getMor()) && group.getMor() ? group.getMorDate() : null);
+
+        dataUse.setMethodsResearch(Objects.nonNull(group.getNmds()) ? !group.getNmds() : null); // TODO: is this right?
```

Comment:
For a dataset, these are the cases as defined in two places, [here](https://github.com/DataBiosphere/consent-ontology/blob/develop/src/main/java/org/broadinstitute/dsde/consent/ontology/datause/services/TextTranslationServiceImpl.java#L212) and [here](https://github.com/DataBiosphere/consent-ontology/blob/develop/src/main/java/org/broadinstitute/dsde/consent/ontology/datause/services/TextTranslationServiceImpl.java#L362). Both look like they use the same logic:

data use methods research: 
- true: "Data use for methods development research irrespective of the specified data use limitations is not prohibited."
- false:
  -  non-null, or falsy: "Data use for methods development research ONLY within the bounds of other data use limitations. [NMDS]"
  - null: "Data use for methods development research irrespective of the specified data use limitations is not prohibited."

Pulling all this together, if our schema has `nmds==true` (which comes from the question `"No Methods Development or validation studies (NMDS)"`), then `dataUse.methodsResearch` should be `false`

However, this doesn't align exactly, so I think we need to bring this point back to Jonathan. We're trying to align a newish version of the NMDS question to the[ V2 algorithm logic](https://github.com/DataBiosphere/consent-ontology/blob/develop/docs/Algorithms.md) and it doesn't exactly match.

Reply:
Sounds good on talking to Jonathan - I am unsure about `NPU` as well. I'll bring it up in standup.